sql/tests: TestRandomSyntaxGeneration failed

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3438840&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3438840&tab=artifacts#/) on master @ [0bc52c216b742e39404ee2404755c3c4acc70b82](https://github.com/cockroachdb/cockroach/commits/0bc52c216b742e39404ee2404755c3c4acc70b82):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x36806c2]

```
Stack: 

```
goroutine 888335 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000a60480, 0x6a96940, 0xc013183bf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x126
panic(0x4bed7e0, 0x8e946b0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil(0xc00aabccb0, 0xc0198ad8d0, 0x8, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521 +0x1e2
github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1(0xc01409dd9c, 0x4, 0xc01409dd01, 0xc00aabccb0, 0x6bbbc00, 0x92e56e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2344 +0x15c
github.com/cockroachdb/cockroach/pkg/sql.forEachRole(0x6a96940, 0xc013183650, 0xc0198ad8d0, 0xc003b3bd68, 0x0, 0xc007905d90)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614 +0x26b
github.com/cockroachdb/cockroach/pkg/sql.glob..func238(0x6a96940, 0xc013183650, 0xc0198ad8d0, 0x6bcc278, 0xc0116a23f0, 0xc004b13c80, 0xc0011d9c38, 0xc007905e7c)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2331 +0xae
github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1(0x69e4b00, 0xc013633560, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533 +0x12a
github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3(0x6a96940, 0xc013183bf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120 +0x154
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2(0xc000a60480, 0x6a96940, 0xc013183bf0, 0xc013183a40, 0x0, 0xc01008a510)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0xf3
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:434 +0x22b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:752: 2m20s of 5m0s: 299778 executions, 2420 successful
E210914 05:21:30.811747 756435 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 30  making txn commit explicit failed for "unnamed" meta={id=03a60f98 pri=0.01420421 epo=0 ts=1631596890.804299914,0 min=1631596890.804299914,0 seq=15} lock=true stat=STAGING rts=1631596890.804299914,0 wto=false gul=1631596891.304299914,0 ifw=14: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m25s of 5m0s: 302481 executions, 2452 successful
E210914 05:21:35.892667 763453 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 31  making txn commit explicit failed for "unnamed" meta={id=50286c97 pri=0.02339864 epo=0 ts=1631596895.884375380,1 min=1631596895.863943950,0 seq=14} lock=true stat=STAGING rts=1631596895.884375380,1 wto=false gul=1631596896.363943950,0 ifw=13: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m30s of 5m0s: 304695 executions, 2477 successful
E210914 05:21:40.896069 768984 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 32  making txn commit explicit failed for "sql txn" meta={id=1961d2a0 pri=0.03972515 epo=0 ts=1631596900.888845831,1 min=1631596896.402776498,0 seq=16} lock=true stat=STAGING rts=1631596900.888845831,1 wto=false gul=1631596896.902776498,0 int=2 ifw=13: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m35s of 5m0s: 306294 executions, 2495 successful
E210914 05:21:45.532405 773776 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 33  making txn commit explicit failed for "unnamed" meta={id=a47642ed pri=0.00964426 epo=0 ts=1631596905.524686404,0 min=1631596905.524686404,0 seq=15} lock=true stat=STAGING rts=1631596905.524686404,0 wto=false gul=1631596906.024686404,0 ifw=14: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m40s of 5m0s: 310535 executions, 2548 successful
E210914 05:21:51.732551 798482 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 34  making txn commit explicit failed for "sql txn" meta={id=6840ad8d pri=0.02493882 epo=0 ts=1631596911.722466441,1 min=1631596911.722205470,0 seq=15} lock=true stat=STAGING rts=1631596911.722466441,1 wto=false gul=1631596912.222205470,0 int=1 ifw=13: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m45s of 5m0s: 318937 executions, 2635 successful
E210914 05:21:56.757779 806633 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 35  making txn commit explicit failed for "sql txn" meta={id=174c01a2 pri=0.08625258 epo=0 ts=1631596916.746357762,1 min=1631596911.981835950,0 seq=15} lock=true stat=STAGING rts=1631596916.746357762,1 wto=false gul=1631596912.481835950,0 int=1 ifw=13: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m50s of 5m0s: 319923 executions, 2650 successful
E210914 05:22:01.042585 809245 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 36  making txn commit explicit failed for "unnamed" meta={id=27286035 pri=0.00031693 epo=0 ts=1631596921.028005825,0 min=1631596921.028005825,0 seq=15} lock=true stat=STAGING rts=1631596921.028005825,0 wto=false gul=1631596921.528005825,0 int=1 ifw=13: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 2m55s of 5m0s: 321827 executions, 2682 successful
    rsg_test.go:752: 3m0s of 5m0s: 329808 executions, 2779 successful
E210914 05:22:08.277486 838729 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 37  making txn commit explicit failed for "sql txn" meta={id=7444ce7d pri=0.00649760 epo=0 ts=1631596928.265737610,0 min=1631596928.265737610,0 seq=15} lock=true stat=STAGING rts=1631596928.265737610,0 wto=false gul=1631596928.765737610,0 ifw=14: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 3m5s of 5m0s: 333754 executions, 2826 successful
E210914 05:22:13.597260 851207 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 38  making txn commit explicit failed for "sql txn" meta={id=ed7c5f13 pri=0.01695221 epo=0 ts=1631596933.587081052,0 min=1631596933.587081052,0 seq=16} lock=true stat=STAGING rts=1631596933.587081052,0 wto=false gul=1631596934.087081052,0 int=1 ifw=14: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 3m10s of 5m0s: 337256 executions, 2856 successful
E210914 05:22:20.895247 884810 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 39  making txn commit explicit failed for "sql txn" meta={id=e7d84082 pri=0.04655994 epo=0 ts=1631596940.883119774,0 min=1631596940.883119774,0 seq=17} lock=true stat=STAGING rts=1631596940.883119774,0 wto=false gul=1631596941.383119774,0 int=1 ifw=15: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:752: 3m15s of 5m0s: 348988 executions, 2983 successful
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40  a panic has occurred!
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +runtime error: invalid memory address or nil pointer dereference
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +(1) attached stack trace
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  -- stack trace:
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | runtime.gopanic
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/usr/local/go/src/runtime/panic.go:965
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | runtime.panicmem
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/usr/local/go/src/runtime/panic.go:212
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | runtime.sigpanic
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/usr/local/go/src/runtime/signal_unix.go:734
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2344
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.forEachRole
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2331
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | runtime.goexit
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +Wraps: (2) runtime error: invalid memory address or nil pointer dereference
E210914 05:22:25.967672 888335 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:42184,hostssl,user=root] 40 +Error types: (1) *withstack.withStack (2) runtime.errorString
```
</p>
</details>
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65680 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-experience branch-release-20.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-kv branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3452147&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3452147&tab=artifacts#/) on master @ [189259e803eca715307bfe0545c84189486a36c4](https://github.com/cockroachdb/cockroach/commits/189259e803eca715307bfe0545c84189486a36c4):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3682a22]

```
Stack: 

```
goroutine 640857 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000bcac00, 0x6a9c118, 0xc01137d650)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x126
panic(0x4befcc0, 0x8e9b6b0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil(0xc006a751b0, 0xc000b7a5d0, 0x8, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521 +0x1e2
github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1(0xc011758dd0, 0x6, 0xc011758d00, 0xc006a751b0, 0x6bc1530, 0x92ede20, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2356 +0x17f
github.com/cockroachdb/cockroach/pkg/sql.forEachRole(0x6a9c118, 0xc01137d290, 0xc000b7a5d0, 0xc005301d58, 0xffffffffffffffff, 0xc000382000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614 +0x26b
github.com/cockroachdb/cockroach/pkg/sql.glob..func238(0x6a9c118, 0xc01137d290, 0xc000b7a5d0, 0x6bd1bb8, 0xc003577200, 0xc00c2e6940, 0xc001233638, 0xc00164067c)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2343 +0xc5
github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1(0x69ea260, 0xc0019d1000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533 +0x12a
github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3(0x6a9c118, 0xc01137d650)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120 +0x154
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2(0xc000bcac00, 0x6a9c118, 0xc01137d650, 0xc01137d4a0, 0x0, 0xc00ef4a030)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0xf3
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:434 +0x22b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
I210916 05:19:24.850180 641946 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4186  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.850675 641920 sql/catalog/lease/descriptor_state.go:139  [n1] 4187  new lease: tabledesc.immutable: {ID: 33, Version: 201, ModificationTime: "1631769564.826523288,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3]}]}
I210916 05:19:24.853028 642006 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4188  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.855182 642007 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4189  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.855357 642032 jobs/adopt.go:247  [-] 4190  job 693695823327821825: resuming execution
I210916 05:19:24.856388 642031 jobs/adopt.go:247  [-] 4191  job 693695823323889665: resuming execution
I210916 05:19:24.857394 642008 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4192  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.857438 642194 jobs/registry.go:1150  [n1] 4193  SCHEMA CHANGE job 693695823327821825: stepping through state running with error: <nil>
I210916 05:19:24.858395 642158 jobs/registry.go:1150  [n1] 4194  SCHEMA CHANGE job 693695823323889665: stepping through state running with error: <nil>
I210916 05:19:24.859450 642009 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4195  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.861517 642010 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4197  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.861530 642194 sql/schema_changer.go:586  [n1,job=693695823327821825,scExec,id=33] 4196  schema change on "role_options" (v201) starting execution...
I210916 05:19:24.861686 642194 sql/schema_changer.go:1024  [n1,job=693695823327821825,scExec,id=33] 4198  waiting for a single version...
I210916 05:19:24.862336 642158 sql/schema_changer.go:586  [n1,job=693695823323889665,scExec,id=4] 4199  schema change on "users" (v174) starting execution...
I210916 05:19:24.862450 642158 sql/schema_changer.go:1024  [n1,job=693695823323889665,scExec,id=4] 4200  waiting for a single version...
I210916 05:19:24.864117 642199 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4201  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.864287 642194 sql/schema_changer.go:1030  [n1,job=693695823327821825,scExec,id=33] 4202  waiting for a single version... done (at v 201)
I210916 05:19:24.864441 642194 jobs/registry.go:1150  [n1] 4203  SCHEMA CHANGE job 693695823327821825: stepping through state succeeded with error: <nil>
I210916 05:19:24.865003 642158 sql/schema_changer.go:1030  [n1,job=693695823323889665,scExec,id=4] 4204  waiting for a single version... done (at v 174)
I210916 05:19:24.865063 642158 jobs/registry.go:1150  [n1] 4205  SCHEMA CHANGE job 693695823323889665: stepping through state succeeded with error: <nil>
I210916 05:19:24.866307 642200 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4206  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
I210916 05:19:24.868434 642201 sql/catalog/lease/descriptor_state.go:239  [n1,client=127.0.0.1:47304,hostssl,user=root,intExec=expand-roles] 4207  release: 23("role_members") ver=1:1631769839.065990052,0, refcount=1
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208  a panic has occurred!
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +runtime error: invalid memory address or nil pointer dereference
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +(1) attached stack trace
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  -- stack trace:
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | runtime.gopanic
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/usr/local/go/src/runtime/panic.go:965
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | runtime.panicmem
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/usr/local/go/src/runtime/panic.go:212
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | runtime.sigpanic
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/usr/local/go/src/runtime/signal_unix.go:734
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2356
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.forEachRole
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2343
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | runtime.goexit
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +Wraps: (2) runtime error: invalid memory address or nil pointer dereference
E210916 05:19:24.868645 640857 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:47304,hostssl,user=root] 4208 +Error types: (1) *withstack.withStack (2) runtime.errorString
```
</p>
</details>
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65680 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-experience branch-release-20.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-kv branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3457473&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3457473&tab=artifacts#/) on master @ [1f0dfa35126af18c678226a77e1848ca489a1bb2](https://github.com/cockroachdb/cockroach/commits/1f0dfa35126af18c678226a77e1848ca489a1bb2):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x36822a2]

```
Stack: 

```
goroutine 83232 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000e6e400, 0x6a9ebd8, 0xc00682a1b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x126
panic(0x4bf1da0, 0x8e9e6b0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil(0xc0032883d0, 0xc0069c85d0, 0x8, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521 +0x1e2
github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1(0xc0054d3410, 0x7, 0xc0054d3400, 0xc0032883d0, 0x6bc41b0, 0x92f0e60, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2360 +0x17c
github.com/cockroachdb/cockroach/pkg/sql.forEachRole(0x6a9ebd8, 0xc007623fb0, 0xc0069c85d0, 0xc0071afd58, 0xffffffffffffffff, 0xc000724000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614 +0x26b
github.com/cockroachdb/cockroach/pkg/sql.glob..func238(0x6a9ebd8, 0xc007623fb0, 0xc0069c85d0, 0x6bd4858, 0xc0054c7dd0, 0xc00150edc0, 0xc000592bb8, 0xc004191e7c)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2345 +0xc5
github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1(0x69ecc60, 0xc003d201c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533 +0x12a
github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3(0x6a9ebd8, 0xc00682a1b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120 +0x154
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2(0xc000e6e400, 0x6a9ebd8, 0xc00682a1b0, 0xc00682a000, 0x0, 0xc006a52270)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0xf3
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:434 +0x22b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
I210917 05:16:49.865561 83725 jobs/adopt.go:247  [-] 692  job 693978430998740993: resuming execution
I210917 05:16:49.867021 83766 jobs/registry.go:1150  [n1] 693  SCHEMA CHANGE job 693978430998740993: stepping through state running with error: <nil>
I210917 05:16:49.867317 83728 jobs/registry.go:1150  [n1] 694  SCHEMA CHANGE job 693978431000936449: stepping through state running with error: <nil>
I210917 05:16:49.870613 83728 sql/schema_changer.go:586  [n1,job=693978431000936449,scExec,id=33] 695  schema change on "role_options" (v33) starting execution...
I210917 05:16:49.870670 83728 sql/schema_changer.go:1024  [n1,job=693978431000936449,scExec,id=33] 696  waiting for a single version...
I210917 05:16:49.872523 83728 sql/catalog/lease/lease.go:139  [n1,job=693978431000936449,scExec,id=33] 697  waiting for 1 leases to expire: desc=[{role_options 33 32}]
I210917 05:16:49.878005 79258 sql/table.go:165  [n1,client=127.0.0.1:59340,hostssl,user=root] 698  queued new schema-change job 693978431088230401 for table 4, mutation 0
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699  a panic has occurred!
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +runtime error: invalid memory address or nil pointer dereference
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +(1) attached stack trace
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  -- stack trace:
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | runtime.gopanic
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/usr/local/go/src/runtime/panic.go:965
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | runtime.panicmem
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/usr/local/go/src/runtime/panic.go:212
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | runtime.sigpanic
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/usr/local/go/src/runtime/signal_unix.go:734
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.roleOptions.validUntil
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2521
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238.1
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2360
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.forEachRole
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2614
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.glob..func238
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:2345
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:533
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:120
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | runtime.goexit
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +Wraps: (2) runtime error: invalid memory address or nil pointer dereference
E210917 05:16:49.878582 83232 1@util/log/logcrash/crash_reporting.go:140  [n1,client=127.0.0.1:59208,hostssl,user=root] 699 +Error types: (1) *withstack.withStack (2) runtime.errorString
I210917 05:16:49.878899 79258 sql/table.go:165  [n1,client=127.0.0.1:59340,hostssl,user=root] 700  queued new schema-change job 693978431091146753 for table 33, mutation 0
I210917 05:16:49.887145 83766 sql/schema_changer.go:586  [n1,job=693978430998740993,scExec,id=4] 701  schema change on "users" (v30) starting execution...
I210917 05:16:49.887274 83766 sql/schema_changer.go:1024  [n1,job=693978430998740993,scExec,id=4] 702  waiting for a single version...
I210917 05:16:49.887572 81903 sql/table.go:165  [n1,client=127.0.0.1:59414,hostssl,user=root] 703  queued new schema-change job 693978431119589377 for table 33, mutation 0
I210917 05:16:49.888335 83493 sql/schema_changer.go:1030  [n1,job=693978430934155265,scExec,id=4] 704  waiting for a single version... done (at v 30)
I210917 05:16:49.888401 83493 jobs/registry.go:1150  [n1] 705  SCHEMA CHANGE job 693978430934155265: stepping through state succeeded with error: <nil>
I210917 05:16:49.889693 83728 sql/schema_changer.go:1030  [n1,job=693978431000936449,scExec,id=33] 706  waiting for a single version... done (at v 33)
I210917 05:16:49.889800 83728 jobs/registry.go:1150  [n1] 707  SCHEMA CHANGE job 693978431000936449: stepping through state succeeded with error: <nil>
I210917 05:16:49.895444 81903 6@util/log/event_log.go:32  [n1,client=127.0.0.1:59414,hostssl,user=root] 708 ={"Timestamp":1631855809849975358,"EventType":"create_role","Statement":"CREATE USER 'char' WITH NOCONTROLCHANGEFEED","Tag":"CREATE ROLE","User":"root","RoleName":"char"}
I210917 05:16:49.895550 81903 jobs/registry.go:345  [n1,client=127.0.0.1:59414,hostssl,user=root] 709  scheduled jobs [693978431025381377 693978431119589377]
I210917 05:16:49.897554 84050 sql/catalog/lease/descriptor_state.go:139  [n1] 710  new lease: tabledesc.immutable: {ID: 4, Version: 31, ModificationTime: "1631855809.887409969,1", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3]}]}
I210917 05:16:49.898851 83766 sql/schema_changer.go:1030  [n1,job=693978430998740993,scExec,id=4] 711  waiting for a single version... done (at v 30)
I210917 05:16:49.898962 83766 jobs/registry.go:1150  [n1] 712  SCHEMA CHANGE job 693978430998740993: stepping through state succeeded with error: <nil>
I210917 05:16:49.899748 84116 sql/catalog/lease/descriptor_state.go:139  [n1] 713  new lease: tabledesc.immutable: {ID: 33, Version: 34, ModificationTime: "1631855809.887409969,1", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3]}]}
I210917 05:16:49.906505 84107 jobs/adopt.go:247  [-] 714  job 693978431119589377: resuming execution
```
</p>
</details>
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65680 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-experience branch-release-20.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-kv branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/information_schema.go
+++ b/pkg/sql/information_schema.go
@@ -2499,38 +2499,29 @@ func (r roleOptions) noLogin() (tree.DBool, error) {
 
 func (r roleOptions) validUntil(p *planner) (tree.Datum, error) {
 	const validUntilKey = "VALID UNTIL"
-	exists, err := r.Exists(validUntilKey)
+	jsonValue, err := r.FetchValKey(validUntilKey)
 	if err != nil {
 		return nil, err
 	}
-
-	var rolValidUntil *time.Time = nil
-	if exists {
-		val, err := r.FetchValKey(validUntilKey)
-		if err != nil {
-			return nil, err
-		}
-		valStr, err := val.AsText()
-		if err != nil {
-			return nil, err
-		}
-
-		validUntil, _, err := pgdate.ParseTimestamp(
-			p.EvalContext().GetRelativeParseTime(),
-			pgdate.DefaultDateStyle(),
-			*valStr,
-		)
-		if err != nil {
-			return nil, errors.Errorf("rolValidUntil string %s could not be parsed with datestyle %s", *valStr, p.EvalContext().GetDateStyle())
-		}
-		rolValidUntil = &validUntil
-	}
-
-	if rolValidUntil == nil {
+	if jsonValue == nil {
 		return tree.DNull, nil
 	}
-
-	return tree.MakeDTimestampTZ(*rolValidUntil, time.Second)
+	validUntilText, err := jsonValue.AsText()
+	if err != nil {
+		return nil, err
+	}
+	if validUntilText == nil {
+		return tree.DNull, nil
+	}
+	validUntil, _, err := pgdate.ParseTimestamp(
+		p.EvalContext().GetRelativeParseTime(),
+		pgdate.DefaultDateStyle(),
+		*validUntilText,
+	)
+	if err != nil {
+		return nil, errors.Errorf("rolValidUntil string %s could not be parsed with datestyle %s", *validUntilText, p.EvalContext().GetDateStyle())
+	}
+	return tree.MakeDTimestampTZ(validUntil, time.Second)
 }
 
 func (r roleOptions) createDB() (tree.DBool, error) {

--- a/pkg/sql/logictest/testdata/logic_test/pg_catalog
+++ b/pkg/sql/logictest/testdata/logic_test/pg_catalog
@@ -5329,3 +5329,15 @@ usename       usesuper
 regular_user  false
 root          true
 super_user    true
+
+# Testing null valid until roles
+statement ok
+CREATE USER regression_70180 WITH password '123' VALID UNTIL null;
+
+query TT colnames
+SELECT usename, valuntil
+FROM pg_user
+WHERE usename = 'regression_70180'
+----
+usename           valuntil
+regression_70180  NULL

